readthedocs/core/middleware.py

```diff
@@ -77,3 +77,38 @@ def __call__(self, request):
             response._csp_update = update_csp_headers[url_name]
 
         return response
+
+
+class LoginMethodCookie:
+    """
+    Set a cookie with the login method used by the user.
+
+    This is used by the templates to put a small "Last used" label next to the method used.
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        response = self.get_response(request)
+        return response
+
+    def process_template_response(self, request, response):
```

Comment:
Don't think you need a middleware for this, this is only needed by the login view. Actually, don't think you need a cookie, since the server isn't doing anything with this, using the local storage should be fine.